docs/contributing.rst

```diff
@@ -66,14 +66,17 @@ Running the compiler tests
 Solidity includes different types of tests. They are included in the application
 called ``soltest``. Some of them require the ``cpp-ethereum`` client in testing mode.
 
-To run ``cpp-ethereum`` in testing mode: ``eth --test -d /tmp/testeth``.
+To run a subset of the tests that do not require ``cpp-ethereum``, use ``./build/test/soltest -- --no-ipc``.
 
-To run the tests: ``soltest -- --ipcpath /tmp/testeth/geth.ipc``.
+For all other tests, you need to install [cpp-ethereum](https://github.com/ethereum/cpp-ethereum/releases/download/solidityTester/eth) and run it in testing mode: ``eth --test -d /tmp/testeth``.
+
+Then you run the actual tests: ``./build/test/soltest -- --ipcpath /tmp/testeth/geth.ipc``.
 
 To run a subset of tests, filters can be used:
 ``soltest -t TestSuite/TestName -- --ipcpath /tmp/testeth/geth.ipc``, where ``TestName`` can be a wildcard ``*``.
 
-Alternatively, there is a testing script at ``scripts/test.sh`` which executes all tests.
+Alternatively, there is a testing script at ``scripts/test.sh`` which executes all tests and runs
+cpp-ethereum automatically if it is in the path (but does not download it).
 
```

Comment:
I'd put a note here that the official testing also runs solc-js and various other javascript tests on the generated emscripten (so that it comes to no surprise if everything else passes but those don't).